sanity/sanity_test.go

```diff
@@ -0,0 +1,290 @@
+		vmiMap:        make(map[string]*kubevirtv1.VirtualMachineInstance),
+		hotpluggedMap: make(map[string]device),
+	}
+	key := getKey(infraClusterNamespace, nodeID)
+	client.vmiMap[key] = &kubevirtv1.VirtualMachineInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nodeID,
+			Namespace: infraClusterNamespace,
+		},
+		Spec: kubevirtv1.VirtualMachineInstanceSpec{
+			Domain: kubevirtv1.DomainSpec{
+				Firmware: &kubevirtv1.Firmware{
+					UUID: nodeID,
+				},
+			},
+		},
+	}
+	return client, client
+}
+
+type devices struct {
+	BlockDevices []device `json:"blockdevices"`
+}
+
+type device struct {
+	SerialID string `json:"serial"`
+	Path     string `json:"path,omitempty"`
+	Name     string `json:"name"`
+	Fstype   string `json:"fstype"`
+}
+
+type fakeKubeVirtClient struct {
+	dvMap         map[string]*cdiv1.DataVolume
+	vmiMap        map[string]*kubevirtv1.VirtualMachineInstance
+	hotpluggedMap map[string]device
+}
+
+func (k *fakeKubeVirtClient) Ping(ctx context.Context) error {
+	return nil
+}
+func (k *fakeKubeVirtClient) ListVirtualMachines(namespace string) ([]kubevirtv1.VirtualMachineInstance, error) {
+	var res []kubevirtv1.VirtualMachineInstance
+	for _, v := range k.vmiMap {
+		if v != nil {
+			res = append(res, *v)
+		}
+	}
+	return res, nil
+}
+
+func (k *fakeKubeVirtClient) DeleteDataVolume(namespace string, name string) error {
+	key := getKey(namespace, name)
+	delete(k.dvMap, key)
+	return nil
+}
+
+func (k *fakeKubeVirtClient) GetDataVolume(namespace string, name string) (*cdiv1.DataVolume, error) {
+	key := getKey(namespace, name)
+	if k.dvMap[key] == nil {
+		return nil, errors.NewNotFound(cdiv1.Resource("DataVolume"), name)
+	}
+	return k.dvMap[key], nil
+}
+
+func (k *fakeKubeVirtClient) CreateDataVolume(namespace string, dataVolume *cdiv1.DataVolume) (*cdiv1.DataVolume, error) {
+	if dataVolume == nil {
+		return nil, fmt.Errorf("Nil datavolume passed")
+	}
+	key := getKey(namespace, dataVolume.Name)
+	dataVolume.SetUID(types.UID(uuid.NewString()))
+	k.dvMap[key] = dataVolume
+	return dataVolume, nil
+}
+func (k *fakeKubeVirtClient) AddVolumeToVM(namespace string, vmName string, hotPlugRequest *kubevirtv1.AddVolumeOptions) error {
+	vmKey := getKey(namespace, vmName)
+	if k.vmiMap[vmKey] == nil {
+		return fmt.Errorf("VM %s/%s not found", namespace, vmName)
+	}
+	k.hotpluggedMap[hotPlugRequest.Name] = device{
+		SerialID: hotPlugRequest.Disk.Serial,
+		Name:     hotPlugRequest.Name,
+		Fstype:   "ext4",
+	}
+	return nil
+}
+
+func (k *fakeKubeVirtClient) RemoveVolumeFromVM(namespace string, vmName string, hotPlugRequest *kubevirtv1.RemoveVolumeOptions) error {
+	vmKey := getKey(namespace, vmName)
+	if k.vmiMap[vmKey] == nil {
+		return fmt.Errorf("VM %s/%s not found", namespace, vmName)
+	}
+	delete(k.hotpluggedMap, hotPlugRequest.Name)
+	return nil
+}
+
+func (m *fakeKubeVirtClient) List() ([]byte, error) {
+	ds := make([]device, 0)
+	for _, value := range m.hotpluggedMap {
```

Comment:
kubevirt as `m`?

Reply:
I had it as a different struct, good catch should be k like the rest.